### DISCUSSION TITLE
what is the correct `@types/react-dom` package now for next@14 (which has useFormStatus)

### DISCUSSION THREAD

**Discussion by stefanprobst** [vercel/next.js#73581] [Help]
### Summary

i have a next v14 project which uses `useFormStatus` imported from `react-dom`:

```ts
import { useFormStatus } from "react-dom";
```

this worked with `@types/react-dom@18.3.1` but it seems the newly released `@types/react-dom@18.3.2` don't expose `useFormStatus` anymore.

what are the correct types packages to use in a next 14 project?

---

to check: run `pnpm create next-app@14` and try to import `useFormStatus`

### Example

pnpm create next-app@14

**Comment by icyJoseph** [comment]
This was added quite a while ago, IIRC: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65338

So like, in the types package version 18.2.4, it'd be `experimental_useFormStatus`, and then some versions later it was bumped to canary, which you could find for example in `18.3.1`, and there it'd be `useFormStatus` only.

**Reply by stefanprobst** [reply]
my point is that 18.3.2 does not seem to export useFormStatus anymore

**Reply by icyJoseph** [reply]
Right, because `useFormStatus` is a React 19 feature, that was made available in the `18` world through canary versioning, but since we have a stable 19 version released now, if you want to use `useFormStatus`, you kind of have to also use 19. 

This whole canary release channel thing was bound to create this kind of confusion, but it also helped advance the development of this features at a faster rate...

**Reply by stefanprobst** [reply]
yep since next v14 AFAIK uses a `react-dom` version which *has* `useFormStatus`, and a `react` version which has `useFormState` but not `useActionState` - the question is what the correct `@types` packages are for that situation.

for now i'll just pin to `@types/react-dom@18.3.1` and `@types/react@18.3.13`

**Reply by stefanprobst** [reply]
@eps1lon will https://github.com/DefinitelyTyped/DefinitelyTyped/pull/71388 help with this situation?

**Reply by eps1lon** [reply]
> @eps1lon will https://github.com/DefinitelyTyped/DefinitelyTyped/pull/71388 help with this situation?

Yes. This was on oversight on my part. For `next@14`, you should continue using `@types/react@18.x`
